### ISSUE TITLE
Block 1, Task 2: First Collaborative Git Steps

### ISSUE THREAD

**Issue by mbannert** [h4sci/h4sci-tasks#2]
- [ ] one person in your group should create a repository called "task2" in their own github account and give access to the other members of the group. Proceed to clone the repository to your local disk
- [ ] Create a markdown file called `task-2-your-name.md` that holds your name, e.g., `task-2-matt-bannert.md`. You may also use an alias instead of your name (as this may end up in a PUBLIC repository), but make sure it's unique within your group.  Please avoid spaces and/or special characters, use lowercase only. Doing so helps avoid encoding problems we might encounter with different locales, different operating systems etc. 
- [ ] add the name of your favorite comic to the file that bears YOUR name.  
- [ ] Add, commit and push the file to the repository. **Make sure to pull before you push**! 
- [ ] Describe your experience in the comments below. Where there any conflicts? If yes, how could they be resolved? 

Reflect on the process, discuss the experience within your team. One team member will have to report your team's experience in class. Make sure to provide your team's ambassador /w a good briefing. Also think about your lab/institute's environment what challenges would you face in such a process?

**Comment by Vilenaria** [comment]
I did not have any problems pushing, but maybe that was since I was the first to work in the repository. I was happy to see that I can pull directly before pushing, i.e. pull, add, commit, pull, push. Are both pulls useful and is it better to pull once too often? 
It might be a good idea to work always with markdown for text-files on git.

**Comment by mbannert** [comment]
Yes, when you are the first, you don't have to expect problems. Nevertheless, you have already done an important step to avoid issues: pull, add, commit, pull, push. That way a conflict that can automatically be resolved would have been resolved and merge (which causes another technical, automatic commit) and then push that. If your friends edited different files you will only get such a merge conflict. 

To answer the question: better pull once more is a good idea.

**Comment by AnitaM123** [comment]
All worked fine :) I tried to clone the repository with the SSH key link, but that did not work, only with the https:// link. Is this about the login with the RSA-key? I don't really understand what is going on there...

**Comment by mbannert** [comment]
Yes @AnitaM123, exactly. There are two ways (=protocols) you can use to talk to github (or other git remote repos): 

- ssh
- SSL  (aka https)

EDIT: Here's our new video on SSH key pairs: https://www.youtube.com/watch?v=yBnPKOcK31A&list=PLurgzfjmXEaOerL2hWo443Asyq0nFyupG&index=4
and book section: https://rse-book.github.io/case-studies.html#sec-rsa


The latter is used by everyone all the time most times we are surfing the web. Back in the days there was http:// which transferred all the info that was exchanged between your computer and some server in plain text. So anybody who put their ears on the line in between could listen to what you talk. By today, the encrypted version of http, namely https is the de facto standard. Basically the same protocol, but encrypted. So whoever listens just sees a bunch of gibberish.
Git allows for https, but offers a second alternative: SSH is less known for everyday use but is very common in server administration and has several advantages over https, particularly when transferring files. 
you should be able to use SSH as well. If you tried it locally on your computer, you need some ssh client to use ssh. On OSX and Linux it should be around, Windows users will have to install a client. 

-> Bottom line: I would not invest tons of time in this now since https works as well. It's not crucial to have this going from the beginning, but thinking about it and being able to explain and see what's going helps certainly to address this at a later stage. That being said, try again 2-3 times and if that doesn't work come back again later.

**Comment by dkoppzh** [comment]
I did not encounter any problems either. However, I have one question: Is there any reason why we had do save our text as a markdown file and not as a R-file or a normal text file?

**Comment by mbannert** [comment]
@dkoppzh : It doesn't really matter at this stage, but markdown is just text with the option to be styled and in turn cause highlighting in good editors, like **bold** for bold as well as other formatting. 

If you use an .R file and just write your name into it, it's obviously not R syntax and the R Studio would mark it softly as an error.
